R/read_csv.R

```diff
@@ -430,16 +430,22 @@ read_csv_metadata <- function(csv_file) {
         inv_metric_split <- strsplit(gsub("# ", "", line), ",")
         if ((length(inv_metric_split) == 0) ||
             ((length(inv_metric_split) == 1) && identical(inv_metric_split[[1]], character(0))) ||
-            grepl("[a-zA-z]", line, perl = TRUE) ||
             inv_metric_split == "#") {
           parsing_done <- TRUE
           parse_key_val <- TRUE
           break;
         }
+        suppressWarnings(
+          numeric_inv_metric_split <- tryCatch(rapply(inv_metric_split, as.numeric), error = function() {
```

Comment:
Just curious, what's the potential error that requires using tryCatch here?

Reply:
In general, the next commented line following the inv metric will always be an empty line and we wont even reach this.

Just want to make this more general so if, for some reason, we add another commented line like

# someval = 123

this will not break. Will actually add a test for that.

Reply:
Ok I see, that makes sense, thanks

Reply:
Though now that I think of it. If its a dense matrix, it will always have rows==columns so we can use that info when parsing.